### ISSUE TITLE
NPE when using the delete-by-query plugin from the TransportClient

### ISSUE THREAD

**Issue by timeu** [elastic/elasticsearch#15693]
I installed the delete-by-query plugin on the elasticsearch server and verified that it works fine using the REST interface:

```JSON
DELETE /index/type/_query?routing=18
{
  "query": { 
    "term": {
      "id": 1558
    }
  }
}
```

However when I try to use the delete-by-query plugin from the `TransportClient`  I get an NPE. 

```JAVA
DeleteByQueryResponse response = new DeleteByQueryRequestBuilder(client, DeleteByQueryAction.INSTANCE)
  .setIndices(index)
  .setTypes(type)
  .setRouting(18)
  .setQuery(QueryBuilders.termQuery("id", id))
.execute().actionGet();
```

Exception: 
```JAVA
java.lang.NullPointerException
	at org.elasticsearch.client.transport.support.TransportProxyClient$1.doWithNode(TransportProxyClient.java:58)
	at org.elasticsearch.client.transport.TransportClientNodesService.execute(TransportClientNodesService.java:212)
	at org.elasticsearch.client.transport.support.TransportProxyClient.execute(TransportProxyClient.java:55)
	at org.elasticsearch.client.transport.TransportClient.doExecute(TransportClient.java:283)
	at org.elasticsearch.client.support.AbstractClient.execute(AbstractClient.java:347)
	at org.elasticsearch.action.ActionRequestBuilder.execute(ActionRequestBuilder.java:85)
	at org.elasticsearch.action.ActionRequestBuilder.execute(ActionRequestBuilder.java:59)
	at org.elasticsearch.action.deletebyquery.DeleteByQueryRequestBuilder.execute(DeleteByQueryRequestBuilder.java:172)
```

I also added `plugin.types=org.elasticsearch.plugin.deletebyquery.DeleteByQueryPlugin` to the `TransportClient` settings.

This is with the 2.1.1 version of the Java API and ES Server

**Comment by timeu** [comment]
Ok I think this is due to [this](https://github.com/elastic/elasticsearch/pull/13055). I will try the `addPlugin`  function and see if it works.

**Comment by timeu** [comment]
Using the `addPlugin` method solves the NPE. 
```JAVA
Client client = TransportClient.builder().settings(settings())
                .addPlugin(DeleteByQueryPlugin.class)
                .build()
```

A better error message would be more helpful for the cases when the user forgets to add a plugin. 
Issue can be closed

**Comment by clintongormley** [comment]
@rjernst would it be possible to throw a more meaningful exception here, rather than an NPE?

**Comment by rjernst** [comment]
We could add an explicit check in 2.x for `plugin.types` in settings when building the transport client, and give an error suggesting to use `addPlugin`? I don't think we should have this indefinitely, but at least for 2.x.

**Comment by clintongormley** [comment]
+1

**Comment by rjernst** [comment]
This was fixed in #15943.

**Comment by NicolasYDDER** [comment]
Hi folks,

I got an NPE with the 2.3 version if `addPlugin` is not define with the `DeleteByQueryPlugin.class`
According this issue it should be fixed, right ?
